ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/src/io/ktor/server/webjars/Webjars.kt

```diff
@@ -1,126 +1,95 @@
-                        lastModified
-                    )
-                )
-            } catch (multipleFiles: MultipleMatchesException) {
-                context.call.respond(HttpStatusCode.InternalServerError)
-            } catch (notFound: IllegalArgumentException) {
             }
         }
-    }
 
     /**
-     * Plugin installation companion.
+     * Makes no effect. Will be dropped in future releases.
      */
-    public companion object Plugin : ApplicationPlugin<ApplicationCallPipeline, Configuration, Webjars> {
-
-        override val key: AttributeKey<Webjars> = AttributeKey("Webjars")
+    @Suppress("unused")
+    @Deprecated("This is no longer used and will be dropped in future releases.", level = DeprecationLevel.ERROR)
+    public var zone: ZoneId = ZoneId.systemDefault()
+}
 
-        override fun install(
-            pipeline: ApplicationCallPipeline,
-            configure: Configuration.() -> Unit
-        ): Webjars {
-            val configuration = Configuration().apply(configure)
+private fun extractWebJar(path: String, knownWebJars: Set<String>, locator: WebJarAssetLocator): String {
+    val firstDelimiter = if (path.startsWith("/")) 1 else 0
+    val nextDelimiter = path.indexOf("/", 1)
+    val webjar = if (nextDelimiter > -1) path.substring(firstDelimiter, nextDelimiter) else ""
+    val partialPath = path.substring(nextDelimiter + 1)
+    if (webjar !in knownWebJars) {
+        throw IllegalArgumentException("jar $webjar not found")
+    }
+    return locator.getFullPath(webjar, partialPath)
+}
 
-            val plugin = Webjars(configuration.path)
+/**
+ * This plugin listens to requests starting with the specified path prefix and responding with static content
+ * packaged into webjars. A [WebJarAssetLocator] is used to look for static files.
+ */
+public val Webjars: ApplicationPlugin<Application, WebjarsConfig, PluginInstance> =
+    createApplicationPlugin("Webjars", ::WebjarsConfig) {
+        val webjarsPrefix = pluginConfig.path
+        require(webjarsPrefix.startsWith("/"))
+        require(webjarsPrefix.endsWith("/"))
 
-            pipeline.intercept(ApplicationCallPipeline.Plugins) {
-                plugin.intercept(this)
+        val locator = WebJarAssetLocator()
+        val knownWebJars = locator.webJars?.keys?.toSet() ?: emptySet()
+        val lastModified = GMTDate()
+
+        onCall { call ->
+            val fullPath = call.request.path()
+            if (fullPath.startsWith(webjarsPrefix) &&
+                call.request.httpMethod == HttpMethod.Get &&
+                fullPath.last() != '/'
+            ) {
+                val resourcePath = fullPath.removePrefix(webjarsPrefix)
+                try {
+                    val location = extractWebJar(resourcePath, knownWebJars, locator)
+                    val stream = WebjarsConfig::class.java.classLoader.getResourceAsStream(location) ?: return@onCall
+                    call.respond(
```

Comment:
Could you add the test when the call is already handled by a different plugin before `Webjars`?